swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/agents/ScheduledAgent.kt

```diff
@@ -75,59 +76,49 @@ abstract class ScheduledAgent(
   }
 
   private fun waitForCacheThenStart() {
-    executorService.schedule({
-      try {
-        while (!cacheStatus.cachesLoaded()) {
-          log.info("Can't work until the caches load...")
-          Thread.sleep(Duration.ofSeconds(5).toMillis())
-        }
-        log.info("Caches loaded.")
-        scheduleSwabbie()
-      } catch (e: Exception) {
-        log.error("Failed while waiting for cache to start in ${javaClass.simpleName}.", e)
+    try {
+      while (!cacheStatus.cachesLoaded()) {
+        log.debug("Can't work until the caches load...")
+        Thread.sleep(Duration.ofSeconds(5).toMillis())
       }
-    }, 15, TimeUnit.SECONDS)
+      log.debug("Caches loaded.")
+      scheduleSwabbie()
+    } catch (e: Exception) {
+      log.error("Failed while waiting for cache to start in ${javaClass.simpleName}.", e)
+    }
   }
 
   private fun scheduleSwabbie() {
-      runningTask = executorService.scheduleWithFixedDelay({
-        try {
-          when {
-            LocalDateTime.now(clock).dayOfWeek in swabbieProperties.schedule.getResolvedDays() ->
-              when {
-                timeToWork(swabbieProperties.schedule, clock) -> {
-                  if (dynamicConfigService.isEnabled("swabbie.work", true)) {
-                    log.info("Swabbie schedule: working")
-                    runSwabbie()
-                  } else {
-                    log.info("Swabbie schedule: disabled via property swabbie.work")
-                  }
+    runningTask = executorService.scheduleWithFixedDelay({
+      when {
+        dynamicConfigService.isEnabled(SWABBIE_FLAG_PROPERY, true) -> {
```

Comment:
This condition seems to be `if swabbie.work == true then do not work`. Based on the log message, this seems to be flipped

Reply:
Yeah good catch. I caught that in the other PR I was working last night. Will update